gtsfm/frontend/verifier/loransac.py

```diff
@@ -0,0 +1,197 @@
+
+        Args:
+            uv_i1: array of shape (N3,2) representing coordinates of 2d points in image 1.
+            uv_i2: array of shape (N3,2) representing coordinates of 2d points in image 2.
+            camera_intrinsics_i1: intrinsics for image #i1.
+            camera_intrinsics_i2: intrinsics for image #i2.
+
+        Returns:
+            dictionary containing result status code, estimated relative pose (R,t), and inlier mask.
+        """
+        focal_length = camera_intrinsics_i1.fx()
+        cx, cy = camera_intrinsics_i1.px(), camera_intrinsics_i1.py()
+
+        # TODO: use more accurate proxy?
+        width = int(cx * 2)
+        height = int(cy * 2)
+
+        camera_dict1 = {
+            "model": "SIMPLE_PINHOLE",
+            "width": width,
+            "height": height,
+            "params": [focal_length, cx, cy],
+        }
+
+        focal_length = camera_intrinsics_i2.fx()
+        cx, cy = camera_intrinsics_i2.px(), camera_intrinsics_i2.py()
+        camera_dict2 = {
+            "model": "SIMPLE_PINHOLE",
+            "width": width,
+            "height": height,
+            "params": [focal_length, cx, cy],
+        }
+        result_dict = pycolmap.essential_matrix_estimation(
+            uv_i1, uv_i2, camera_dict1, camera_dict2, max_error_px=self._estimation_threshold_px
+        )
+        return result_dict
+
+    def verify(
+        self,
+        keypoints_i1: Keypoints,
+        keypoints_i2: Keypoints,
+        match_indices: np.ndarray,
+        camera_intrinsics_i1: Cal3Bundler,
+        camera_intrinsics_i2: Cal3Bundler,
+    ) -> Tuple[Optional[Rot3], Optional[Unit3], np.ndarray, float]:
+        """Performs verification of correspondences between two images to recover the relative pose and indices of
+        verified correspondences.
+
+        Args:
+            keypoints_i1: detected features in image #i1.
+            keypoints_i2: detected features in image #i2.
+            match_indices: matches as indices of features from both images, of shape (N3, 2), where N3 <= min(N1, N2).
+            camera_intrinsics_i1: intrinsics for image #i1.
+            camera_intrinsics_i2: intrinsics for image #i2.
+
+        Returns:
+            Estimated rotation i2Ri1, or None if it cannot be estimated.
+            Estimated unit translation i2Ui1, or None if it cannot be estimated.
+            Indices of verified correspondences, of shape (N, 2) with N <= N3. These are subset of match_indices.
+            Inlier ratio of w.r.t. the estimated model, i.e. the #final RANSAC inliers/ #putatives.
+        """
+        if match_indices.shape[0] < self._min_matches:
+            logger.info("[LORANSAC] Not enough correspondences for verification.")
+            return self._failure_result
+
+        uv_i1 = keypoints_i1.coordinates
+        uv_i2 = keypoints_i2.coordinates
+
+        uv_i1_ = uv_i1[match_indices[:, 0]]
+        uv_i2_ = uv_i2[match_indices[:, 1]]
```

Comment:
do it in single step? We will lose some readability but having very similar names would create more confusion?

Reply:
i cleaned it up -- let me know if it's clearer now